### ISSUE TITLE
Add option to filter incoming email

### ISSUE THREAD

**Issue by garrettr** [C3S/redmine_openpgp#9]
Currently, redmine_openpgp allows you to configure settings for _outgoing_ email. If a user does not have a public key set in their account, you can configure redmine_openpgp to filter, block, or send mail to them anyway (unencrypted). If the mail is filtered, the plugin sends a nice email that explains why the content was not included and advises them to configure a public key for their account.

It would be nice to have a similar option for _incoming_ email. The use case here is that we want to remind users that all email they send to the server should be encrypted to protect it in transit. If they send an email (e.g. to create a new ticket or to reply to an ongoing thread) unencrypted, it should be possible to configure redmine_openpgp to filter these, meaning it will reply and explain that this server requires encrypted email and they should configure a public key for their account. This will not protect that first email (which was already sent unencrypted, nothing we can do about that), but should hopefully remind them ASAP to configure their account correctly.

I propose that we add an "unencrypted emails" drop-down menu to Administration -> Plugins -> OpenPGP under the "Incoming settings" with the following options:
- filtered - the email is not processed, and Redmine sends a reply explaining that they should configure their public key
- unchanged - unencrypted incoming emails are allowed without warning (this is the de facto current setting)

**Comment by timegrid** [comment]
Agreed. This seems quite easy and would enhance the user experience.

It would also be nice to reply with a special message, if the user key expired (we had this usecase already several times).
